dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/QuayImageRegistry.java

```diff
@@ -107,20 +115,33 @@ public QuayImageRegistry(final Token quayToken) {
         final List<Tag> tags = new ArrayList<>();
         final Optional<QuayRepo> toolFromQuay = getToolFromQuay(tool);
         if (toolFromQuay.isPresent()) {
-            final QuayRepo quayRepo = toolFromQuay.get();
-            final Map<String, QuayTag> tagsFromRepo = quayRepo.getTags();
-            final int maxQuayTagsReturnedByRepo = 500;
-            List<QuayTag> quayTags = new ArrayList<>(tagsFromRepo.values());
-            if (tagsFromRepo.size() == maxQuayTagsReturnedByRepo) {
-                try {
-                    quayTags = getAllQuayTags(repo);
-                } catch (ApiException e) {
-                    throw new CustomWebApplicationException("Could not get QuayTag", HttpStatus.SC_INTERNAL_SERVER_ERROR);
-                }
+            List<QuayTag> quayTags;
+            try {
+                quayTags = getAllQuayTags(repo);
+            } catch (ApiException e) {
+                throw new CustomWebApplicationException("Could not get QuayTag", HttpStatus.SC_INTERNAL_SERVER_ERROR);
             }
-            for (QuayTag tagItem : quayTags) {
+
+            // Search through the Quay tags for ones that are classified as a manifest list and then use its digest to get the repo's manifest.
+            List<QuayTag> cleanedQuayTagList = new ArrayList<>(quayTags);
+            Map<QuayTag, List<Image>> multiImageQuayTags = new HashMap<>();
+            Gson g = new Gson();
+            quayTags.stream().forEach(quayTag -> {
+                if (quayTag.isIsManifestList() != null && quayTag.isIsManifestList()) {
```

Comment:
Optionally, this is fine as you have it: `Boolean.TRUE.equals(quayTag.isIsManifestList())`